File: 第10章 逻辑推理/10.4 国王的保镖/国王的保镖.py
'''
程序：国王的保镖
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
'''
def judge(who):
    '''判断谁击中了杀手'''
    a = who == 8 or who == 6
    b = who == 2
    c = who == 7
    d = who != 2
    e = not (who == 8 or who == 6)
    f = not (who == 6 or who == 8)
    g = who != 3
    h = who == 8 or who == 6
    if a + b + c + d + e + f + g + h == 3:
        print(who)

def main():
    '''国王的保镖'''
    who = 1
    while who <= 8:
        judge(who)
        who += 1

if __name__ == '__main__':
    main()

File: 第10章 逻辑推理/10.4 国王的保镖/国王的保镖.cpp
/*
程序：谁击中了杀手 
作者：苏秦@小海豚科学馆公众号
来源：图书《趣味编程三剑客：从Scratch迈向Python和C++》
*/
#include <bits/stdc++.h>
using namespace std;

// 判断谁击中了杀手 
void judge(int who)
{
	int a = who == 8 or who == 6;
	int b = who == 2;
	int c = who == 7;
	int d = who != 2;
	int e = not (who == 8 or who == 6);
	int f = not (who == 8 or who == 6);
	int g = who != 3;
	int h = who == 8 or who == 6;
	if (a + b + c + d + e + f + g + h == 3)
		cout << who << endl;
}

// 国王的保镖 
int main()
{
	int who = 1;
	while (who <= 8) {
		judge(who);
		who += 1;
	}
	return 0;
}
